platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileIteratorPbj.java

```diff
@@ -0,0 +1,271 @@
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.time.Instant;
+import java.util.Objects;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+/**
+ * Iterator class for iterating over data items in a DataFile. It is designed to be used in a while(iter.next()){...}
+ * loop and you can then read the data items info for current item with getDataItemsKey, getDataItemsDataLocation and
+ * getDataItemData.
+ *
+ * It is designed to be used from a single thread.
+ *
+ * @see DataFileWriter for definition of file structure
+ */
+@SuppressWarnings("rawtypes")
+public final class DataFileIteratorPbj implements DataFileIterator {
+    /**
+     * Since {@code com.swirlds.platform.Browser} populates settings, and it is loaded before any
+     * application classes that might instantiate a data source, the {@link ConfigurationHolder}
+     * holder will have been configured by the time this static initializer runs.
+     */
+    private static final MerkleDbConfig config = ConfigurationHolder.getConfigData(MerkleDbConfig.class);
+
+    /** Input stream this iterator is reading from */
+    private final BufferedInputStream inputStream;
+    /** Readable data on top of the input stream */
+    private final ReadableSequentialData in;
+    /** The file metadata read from the end of file */
+    private final DataFileMetadata metadata;
+    /** The path to the file we are iterating over */
+    private final Path path;
+    /** The serializer used for reading data from the file */
+    private final DataItemSerializer dataItemSerializer;
+
+    /** Buffer that is reused for reading each data item */
+    private BufferedData dataItemBuffer;
+    /** Index of current data item this iterator is reading, zero being the first item, -1 being before start */
+    private long currentDataItem = -1;
+    /** The offset in bytes from start of file to the beginning of the current item. */
+    private long currentDataItemFilePosition = 0;
+    /** The size in bytes of the current data item */
+    private int currentDataItemSize = 0;
+    /** True if this iterator has been closed */
+    private boolean closed = false;
+
+    /**
+     * Create a new DataFileIterator on an existing file.
+     *
+     * @param path
+     * 		The path to the file to read.
+     * @param metadata
+     * 		The metadata read from the file.
+     * @throws IOException
+     * 		if there was a problem creating a new InputStream on the file at path
+     */
+    public DataFileIteratorPbj(
+            final Path path, final DataFileMetadata metadata, final DataItemSerializer dataItemSerializer)
+            throws IOException {
+        this.path = path;
+        this.metadata = metadata;
+        this.dataItemSerializer = dataItemSerializer;
+        /* FUTURE WORK - https://github.com/swirlds/swirlds-platform/issues/3929 */
```

Comment:
This comment belongs to `MerkleDbConfig` class because it's about the size of the buffer we use.

Reply:
OK, I removed it from `DataFileIteratorJdb` and `DataFileIteratorPbj` and placed to `MerkleDbConfig` The link is also updated from the old swirlds/swirlds-platform github repo to the new hashgraph/hedera-services one: https://github.com/hashgraph/hedera-services/issues/5178.